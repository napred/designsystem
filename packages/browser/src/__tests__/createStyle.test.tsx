import React from 'react';
import { render } from 'react-testing-library';

import { createComponent, createStyle, css, DesignSystem } from '../';

const Box = createComponent('Box', 'div');

describe('createStyle', () => {
  it('works with simple string', () => {
    const styles = [createStyle(['test'], 'color: blue;')];
    const { asFragment } = render(
      <DesignSystem styles={styles}>
        <Box />
      </DesignSystem>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('works with simple style', () => {
    const styles = [
      createStyle(
        ['test'],
        css`
          color: blue;
        `,
      ),
    ];
    const { asFragment } = render(
      <DesignSystem styles={styles}>
        <Box />
      </DesignSystem>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('works with simple object', () => {
    const styles = [createStyle(['test'], { color: 'blue' })];
    const { asFragment } = render(
      <DesignSystem styles={styles}>
        <Box />
      </DesignSystem>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('works with function', () => {
    const styles = [
      createStyle(
        ['test'],
        (props: { color: string }) =>
          css`
            color: ${props.color};
          `,
      ),
    ];
    const { asFragment } = render(
      <DesignSystem styles={styles}>
        <Box color="blue" />
      </DesignSystem>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
