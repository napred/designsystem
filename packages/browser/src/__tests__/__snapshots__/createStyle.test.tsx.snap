// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createStyle works with function 1`] = `
<DocumentFragment>
  .css-ng7bw9 {
  color: #3B5998;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  color: blue;
}

<div
    class="css-ng7bw9"
  />
</DocumentFragment>
`;

exports[`createStyle works with simple object 1`] = `
<DocumentFragment>
  .css-1f227de {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  color: blue;
}

<div
    class="css-1f227de"
  />
</DocumentFragment>
`;

exports[`createStyle works with simple string 1`] = `
<DocumentFragment>
  .css-14q17fo {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  color: blue;
}

<div
    class="css-14q17fo"
  />
</DocumentFragment>
`;

exports[`createStyle works with simple style 1`] = `
<DocumentFragment>
  .css-8xq7br {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  color: blue;
}

<div
    class="css-8xq7br"
  />
</DocumentFragment>
`;
